packages/web_ui/src/components/SlavesPage.jsx

```diff
@@ -50,30 +66,105 @@ function GenerateSlaveTokenButton(props) {
 				setToken(null);
 				form.resetFields();
 			}}
+			width="700px"
 		>
 			<Form form={form} layout="vertical" requiredMark="optional">
 				<Form.Item name="slaveId" label="Slave ID">
-					<Input/>
-				</Form.Item>
-				<Form.Item>
-					<Button
-						onClick={() => { generateToken().catch(notifyErrorHandler("Error generating token")); }}
-					>Generate</Button>
-				</Form.Item>
-				<Form.Item label="Token" required>
-					<Input.TextArea value={token} autoSize ref={tokenTextAreaRef} />
+					<Input onChange={() => { generateToken().catch(notifyErrorHandler("Error generating token")); }} />
 				</Form.Item>
```

Comment:
Slave ID is marked optional yet there's no way to generate an input without inputting an ID here.

![image](https://user-images.githubusercontent.com/14362102/179361858-3bf82b2a-5fe4-4a0d-baa9-2459bf77fe05.png)